### DISCUSSION TITLE
Create Email Required API parameters

### DISCUSSION THREAD

**Discussion by jdhitsolutions** [buttondown/discussions#27] [Q&A] [answered]
I've started testing the API to create emails. I am hoping to build something that will let me post the draft, and then I can review and schedule it manually in the web UI. However, I am getting an error about a missing required field. But, it doesn't tell me which one. Or can someone post a curl example?

**Comment by jmduke** [comment, marked as answer]
A `POST` to `/v1/emails` by default should return the following:

```
{'detail': [{'loc': ['body', 'payload'],
   'msg': 'field required',
   'type': 'value_error.missing'}]}
```

(Which means the "body" is missing.) Try plopping that in! (Or, if you are, share the curl!)

**Reply by jdhitsolutions** [reply]
Does the body need to be formatted as JSON? I assumed I could join parameters like this:

```shell
https://api.buttondown.email/v1/emails?subject=My Subject&body=this is a sample email body&status=draft&email_type=private
```

But I don't know how this approach works when the body will be a 1500-word markdown document. That's why I think I'm missing a step.

**Reply by jmduke** [reply]
Yup, you'll need to JSON-encode the payload. Are you using cURL or Powershell (or something else)?

**Reply by jdhitsolutions** [reply]
I'm using PowerShell. Creating the JSON is easy. Is the body the payload, or all of the parameters and their values? That's why I was looking for a curl template or example. Maybe a developer could look at your docs and understand that JSON encoding was required, but I didn't make that connection.

**Reply by jdhitsolutions** [reply]
I think I have it now. I'm confusing the body of the email with the body of the payload. Let me try something.

**Reply by jdhitsolutions** [reply]
I'm over the hurdle once I got my bodies in order.

**Reply by jdhitsolutions** [reply]
I would suggest making a note in the documentation that the default action is to publish. I forgot to set the status to draft and my test email went out to all subscribers. Oops. I'd prefer that the default status be Draft or make the status a mandatory parameter. You don't want people, like me, learning to use the API and accidentally sending emails.

**Reply by jmduke** [reply]
Yes, definitely! Will call that out in the endpoint docs.
